portality/models/bibjson.py

```diff
@@ -98,13 +98,16 @@ def keywords(self):
         return self._get_list("keywords")
 
     def add_keyword(self, keyword):
-        self._add_to_list_with_struct("keywords", keyword)
+        self._add_to_list_with_struct("keywords", keyword.lower())
 
     def set_keywords(self, keywords):
-        lowercase_keywords = []
-        for k in keywords:
-            lowercase_keywords.append(k.lower())
-        self._set_with_struct("keywords", lowercase_keywords)
+        if type(keywords) is list:
+            lowercase_keywords = []
+            for k in keywords:
+                lowercase_keywords.append(k.lower())
```

Comment:
This is fine, but you could save lines and memory by using:
`keywords = map(str.lower, keywords)`
**or**
`keywords = [w.lower() for l in keywords]`

Both apply the transformation in-place. In this case it's very minor since we restrict keyword length!

Reply:
Cool! I'm gonna change it to remember next time ;)

Reply:
@amdomanska good choice with the list comprehension - @richard-jones gets uncomfortable when I use functional programming like `map` and `fold` :stuck_out_tongue: